components/collapsible-panel/collapsible-panel.js

```diff
@@ -436,11 +454,15 @@ class CollapsiblePanel extends SkeletonMixin(FocusMixin(RtlMixin(LitElement))) {
 	}
 
 	_onBlur() {
-		this._focused = false;
-		this._clicked = false;
+		setTimeout(() => {
+			// don't remove focus if the button still ends up in focus
+			if (getComposedActiveElement() === this.shadowRoot.querySelector('button')) return;
+			this._focused = false;
+		}, 10);
```

Comment:
This is for the scenario of `tab` into the header (focusing on button) > click on the header
The focus outline remains when the click occurs, which looks to be consistent with other usages of focus-visible (e.g., switch, button). Let me know if this logic is incorrect.

Reply:
I think Dave's asking about whether we could totally switch the approach we're using for the focus styles by leveraging _only_ `:has(:focus-visible)` to apply the outline instead of tracking `_focused` and using that. In theory, only tabbing to the `<button>` should trigger `:has(:focus-visible)` and not any clicking. In practice you may have discovered otherwise though...

Reply:
Ah yup sadly that doesn't seem to work.